File: examples/typescript/_shared.ts
// Shared bootstrap. Every example imports `client` + the model constants
// from here so the Fable-specific code stays front-and-centre.
import "dotenv/config";
import Anthropic from "@anthropic-ai/sdk";

// Use the exact model string — no date suffix.
export const MODEL = "claude-fable-5";

// Where to send work if Fable's safety classifiers decline a request.
export const FALLBACK_MODEL = "claude-opus-4-8";

if (!process.env.ANTHROPIC_API_KEY) {
  console.error(
    "ANTHROPIC_API_KEY is not set.\n" +
      "  export ANTHROPIC_API_KEY='sk-ant-...'   (or put it in a .env file)\n" +
      "Note: your org must have 30-day data retention enabled — Fable 5 is\n" +
      "not available under zero-data-retention and will 400 on every request.",
  );
  process.exit(1);
}

// Reads ANTHROPIC_API_KEY from the environment automatically.
export const client = new Anthropic();

File: examples/python/_shared.py
"""Shared bootstrap for the Fable examples.

Every example imports `client` and the model constants from here so the
interesting, Fable-specific code stays front-and-centre in each file.
"""

import os

from anthropic import Anthropic

# Optional: load a local .env so you don't have to export the key each time.
try:
    from dotenv import load_dotenv

    load_dotenv()
except ImportError:
    pass

# The model this whole repo is about. Use the exact string — no date suffix.
MODEL = "claude-fable-5"

# Where to send work if Fable's safety classifiers decline a request.
FALLBACK_MODEL = "claude-opus-4-8"

if not os.getenv("ANTHROPIC_API_KEY"):
    raise SystemExit(
        "ANTHROPIC_API_KEY is not set.\n"
        "  export ANTHROPIC_API_KEY='sk-ant-...'   (or put it in a .env file)\n"
        "Note: your org must have 30-day data retention enabled — Fable 5 is\n"
        "not available under zero-data-retention and will 400 on every request."
    )

# Reads ANTHROPIC_API_KEY from the environment automatically.
client = Anthropic()
